jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/info/ThreadGroupInfo.java

```diff
@@ -0,0 +1,207 @@
+	static final String THREAD_GROUP_PREFIX = "threadgroup."; //$NON-NLS-1$
+	static final String CPU_TIME = "cpu_time"; //$NON-NLS-1$
+	static final String USER_TIME = "user_time"; //$NON-NLS-1$
+	private static final String THREAD_GROUP_VERSION = THREAD_GROUP_PREFIX + "version"; //$NON-NLS-1$
+	private static final String THREAD_GROUP_JAVA_INFO = THREAD_GROUP_PREFIX + "java_info"; //$NON-NLS-1$
+	private static final String NUM_THREADS = THREAD_GROUP_PREFIX+"num_threads"; //$NON-NLS-1$
+	private static ThreadMXBean threadBean;
+	
+	private final int numThreads;
+	private final JvmThreadInfo[] myThreads;
+	private final String groupInfoVersion;
+	private final DiagnosticProperties myProps;
+	private final String javaInfo;
+	private boolean addSynchronizers;
+	
+	/**
+	 * Create an object from received properties
+	 * @param props Properties receive from the target
+	 * @param printSynchronizers add ownable synchronizers to output
+	 * @throws IOException if properties file is corrupt
+	 */
+	public ThreadGroupInfo(Properties props, boolean printSynchronizers) throws  IOException  {
+		addSynchronizers = printSynchronizers;
+		try {
+			DiagnosticProperties.dumpPropertiesIfDebug("Thread group:", props); //$NON-NLS-1$
+			myProps = new DiagnosticProperties(props);
+			if (myProps.containsField(IPC.PROPERTY_DIAGNOSTICS_ERROR) && myProps.getBoolean(IPC.PROPERTY_DIAGNOSTICS_ERROR)) {
+				final String targetMsg = myProps.getProperty(IPC.PROPERTY_DIAGNOSTICS_ERRORMSG);
+				String msg = myProps.getProperty(IPC.PROPERTY_DIAGNOSTICS_ERRORTYPE); 
+				if ((null != targetMsg) && !targetMsg.isEmpty()) {
+					msg = msg+ ": " +  targetMsg; //$NON-NLS-1$
+				}
+				throw new IOException(msg);
+			}
+			numThreads = myProps.getInt(NUM_THREADS);
+			groupInfoVersion = props.getProperty(THREAD_GROUP_VERSION);
+			javaInfo = props.getProperty(THREAD_GROUP_JAVA_INFO);
+
+			if (numThreads <= 0) {
+				throw new UnsupportedOperationException("Cannot determine number of threads"); //$NON-NLS-1$
+			}
+			myThreads = new JvmThreadInfo[numThreads];
+			for (int i = 0; i < numThreads; ++i) {
+				myThreads[i] = new JvmThreadInfo(myProps, i);
+			}
+		} catch (NumberFormatException e) {
+			throw new IOException(e);
+		}
+	}
+
+	/**
+	 * Acquire stack traces for all running threads
+	 * @return Properties objects thread information
+	 */
+	public static DiagnosticProperties getThreadInfoProperties() {
+		threadBean = ManagementFactory.getThreadMXBean();
+		ThreadInfo infos[] = threadBean.dumpAllThreads(true, true);
+		DiagnosticProperties props = getThreadInfoPropertiesImpl(infos);
+		return props;
+	}
+
+	/**
+	 * Acquire stack traces for selected threads
+	 * @param threadIds list of threads to query
+	 * @return Properties objects thread information
+	 * @note some entries my be null if a specified thread no longer exists
+	 */
+	public static DiagnosticProperties getThreadInfoProperties(long[] threadIds) {
```

Comment:
There was no reply to this comment.

Reply:
I would like to leave it as-is because:
- it simplifies a unit test which works on a known thread by not having to filter out the other threads
- It allows other utilities which would work on s specific thread or set of threads
- The extra no-argument variant of this method method is a wrapper and there is no duplicated code.